### PR TITLE
5419 get specific data universe

### PR DESCRIPTION
**Pull request by konangc87**
# Select specific rows based on custom filters.

## Issue
[5419](https://github.com/wazuh/wazuh/issues/5419)

## Description
Select specific rows based on custom filters defined by the user of the library.

This would be performed in the actual database (without any interaction with temp tables).

## Tasks
- [X] Implementation
- [X] Unit tests
image

- [X] CPPCHECK
image

- [X] Valgrind
- [X] Add action and cases to dbsync_test_tool
image

### CODE REVIEW

File: src/dbsync/src/dbsync.cpp [hunk 1]
```diff
@@ -334,12 +334,55 @@ int dbsync_sync_row(const DBSYNC_HANDLE handle,
...
-int dbsync_select_rows(const DBSYNC_HANDLE /*handle*/,
-                       const cJSON*        /*js_data_input*/,
-                       callback_data_t     /*callback_data*/)
+int dbsync_select_rows(const DBSYNC_HANDLE handle,
+                       const cJSON*        js_data_input,
+                       callback_data_t     callback_data)
 {
-    // Dummy function for now.
-    return 0;
+    auto ret_val { -1 };
```
Comment (tinachoB) [inline src/dbsync/src/dbsync.cpp:341]: retVal

File: src/dbsync/src/dbsync.cpp [hunk 2]
```diff
@@ -334,12 +334,55 @@ int dbsync_sync_row(const DBSYNC_HANDLE handle,
...
+        {
+            errorMessage += "json error, id: " + std::to_string(ex.id) + ". " + ex.what();
+            ret_val = ex.id;
+        }
+        catch(const DbSync::dbsync_error& ex)
+        {
+            errorMessage += "DB error, id: " + std::to_string(ex.id()) + ". " + ex.what();
+            ret_val = ex.id();
+        }
+        catch(const DbSync::max_rows_error& ex)
```
Comment (tinachoB) [inline src/dbsync/src/dbsync.cpp:373]: catch block not needed

File: src/dbsync/src/sqlite/sqlite_dbengine.cpp [hunk 3]
```diff
@@ -254,6 +254,69 @@ void SQLiteDBEngine::returnRowsMarkedForDelete(const nlohmann::json& tableNames,
...
+    const auto& sql{ buildSelectQuery(table, query) };
+    auto const& stmt { getStatement(sql) };
+    const auto tableFields { m_tableFields[table] };
+
+    while (SQLITE_ROW == stmt->step())
+    {
+        Row row;
+        const auto& columns{ query.at("column_list") };
+        if (std::find_if(columns.begin(), columns.end(),[](const std::string& value){return value == "*";}) != columns.end())
+        {
```
Comment (tinachoB) [inline src/dbsync/src/sqlite/sqlite_dbengine.cpp:270]: Please, add a brief comment saying why we need to do this if/else (* vs specific columns)

File: src/dbsync/src/sqlite/sqlite_dbengine.cpp [hunk 4]
```diff
@@ -254,6 +254,70 @@ void SQLiteDBEngine::returnRowsMarkedForDelete(const nlohmann::json& tableNames,
     }
 }
 
+void SQLiteDBEngine::selectData(const std::string& table,
+                                const nlohmann::json& query,
+                                const DbSync::ResultCallback& callback)
+{
+    const auto& stmt { getStatement(buildSelectQuery(table, query)) };
```
Comment (konangc87) [inline src/dbsync/src/sqlite/sqlite_dbengine.cpp:261]: add loadTable.

**Pull request merged**